modules/react/select/stories/stories_VisualTesting.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+
+import {
+  ComponentStatesTable,
+  permutateProps,
+  StaticStates,
+} from '@workday/canvas-kit-react/testing';
+import {withSnapshotsEnabled, customColorTheme} from '../../../../utils/storybook';
+import {FormField} from '@workday/canvas-kit-react/form-field';
+
+import {Select, useSelectModel} from '@workday/canvas-kit-react/select';
+
+export default withSnapshotsEnabled({
+  title: 'Testing/Inputs/Select',
+  component: Select,
+});
+
+const options = [
+  {id: 'E-mail', data: {textValue: 'foo'}},
+  {id: 'Phone'},
+  {id: 'Fax (disabled)', disabled: true},
+  {id: 'Mail'},
+  {id: 'Mobile Phone'},
+  {
+    id: 'The Ontologically Anthropocentric Sensory Immersive Simulation',
+    disabled: false,
+  },
+];
+
+const disabledItems = options.filter(item => item.disabled === true).map(item => item.id);
+
+export const SelectStates = () => {
+  const model = useSelectModel({
+    items: options,
+    nonInteractiveIds: disabledItems,
+  });
+  return (
+    <StaticStates>
+      <ComponentStatesTable
+        rowProps={[
+          {label: 'Default', props: {}},
+          {label: 'Alert', props: {error: FormField.ErrorType.Alert}},
+          {label: 'Error', props: {error: FormField.ErrorType.Error}},
+        ]}
+        columnProps={permutateProps(
+          {
+            className: [
+              {label: 'Default', value: ''},
+              {label: 'Hover', value: 'hover'},
+              {label: 'Focus', value: 'focus'},
+              {label: 'Focus Hover', value: 'focus hover'},
+              {label: 'Active', value: 'active'},
+              {label: 'Active Hover', value: 'active hover'},
+            ],
+            disabled: [
+              {label: '', value: false},
+              {label: 'Disabled', value: true},
+            ],
+          },
+          props => {
+            if (props.disabled && !['', 'hover'].includes(props.className)) {
+              return false;
+            }
+            return true;
+          }
```

Comment:
```suggestion
          props => {
            return !props.disabled || !props.className || props.className === 'hover';
          }
```
This conditional will always return true or false, false will be for all items that disabled with class name hover